src/native/eventpipe/ds-eventpipe-protocol.c

```diff
@@ -415,32 +452,33 @@ eventpipe_protocol_helper_stop_tracing (
 static
 bool
 eventpipe_protocol_helper_collect_tracing (
-	DiagnosticsIpcMessage *message,
+	EventPipeCollectTracingCommandPayload *payload,
 	DiagnosticsIpcStream *stream)
 {
-	ep_return_false_if_nok (message != NULL && stream != NULL);
+	ep_return_false_if_nok (stream != NULL);
 
+	EventPipeSessionID session_id = 0;
 	bool result = false;
-	EventPipeCollectTracingCommandPayload *payload;
-	payload = (EventPipeCollectTracingCommandPayload *)ds_ipc_message_try_parse_payload (message, eventpipe_collect_tracing_command_try_parse_payload);
 
 	if (!payload) {
 		ds_ipc_message_send_error (stream, DS_IPC_E_BAD_ENCODING);
 		ep_raise_error ();
 	}
 
-	EventPipeSessionID session_id;
-	session_id = ep_enable (
-		NULL,
-		payload->circular_buffer_size_in_mb,
-		dn_vector_data_t (payload->provider_configs, EventPipeProviderConfiguration),
-		dn_vector_size (payload->provider_configs),
-		EP_SESSION_TYPE_IPCSTREAM,
-		payload->serialization_format,
-		true,
-		ds_ipc_stream_get_stream_ref (stream),
-		NULL,
-		NULL);
+	EventPipeSessionOptions options = EventPipeSessionOptions();
+	options.output_path = NULL;
+	options.circular_buffer_size_in_mb = payload->circular_buffer_size_in_mb;
+	options.providers = dn_vector_data_t (payload->provider_configs, EventPipeProviderConfiguration);
+	options.providers_len = dn_vector_size (payload->provider_configs);
+	options.session_type = EP_SESSION_TYPE_IPCSTREAM;
+	options.format = payload->serialization_format;
+	options.rundown_requested = payload->rundown_requested;
+	options.disable_stacktrace = payload->disable_stacktrace;
+	options.stream = ds_ipc_stream_get_stream_ref (stream);
+	options.sync_callback = NULL;
+	options.callback_additional_data = NULL;
+
+	session_id = ep_enable_3(options);
```

Comment:
I'm totally fine either way since you already changed it and the perf doesn't matter, but typically we'd pass a large struct by address rather than by value. (as in `ep_enable_3(&options)`)

Reply:
Done!